userspace/libsinsp/sinsp_events.h

```diff
@@ -0,0 +1,369 @@
+	 * @return true if the event type has at least one of these flags.
+ */
+bool is_unused_event(ppm_event_code event_type);
+
+/**
+	 * @brief If the event type has one of the following flags return true:
+	 * - `EF_SKIPPARSERESET`
+	 *
+	 * @param event_type type of event we want to check (must be less than `PPM_EVENT_MAX`)
+	 * @return true if the event type has at least one of these flags.
+ */
+bool is_skip_parse_reset_event(ppm_event_code event_type);
+
+/**
+	 * @brief Return true if the event has the `EF_OLD_VERSION` flag
+	 *
+	 * @param event_type type of event we want to check (must be less than `PPM_EVENT_MAX`)
+	 * @return true if the event type has the `EF_OLD_VERSION` flag.
+ */
+bool is_old_version_event(ppm_event_code event_type);
+
+/**
+	 * @brief Return true if the event belongs to the `EC_SYSCALL` category
+	 *
+	 * @param event_type type of event we want to check (must be less than `PPM_EVENT_MAX`)
+	 * @return true if the event type has the `EC_SYSCALL` category.
+ */
+bool is_syscall_event(ppm_event_code event_type);
+
+/**
+	 * @brief Return true if the event belongs to the `EC_TRACEPOINT` category
+	 *
+	 * @param event_type type of event we want to check (must be less than `PPM_EVENT_MAX`)
+	 * @return true if the event type has the `EC_TRACEPOINT` category.
+ */
+bool is_tracepoint_event(ppm_event_code event_type);
+
+/**
+	 * @brief Return true if the event belongs to the `EC_METAEVENT` category
+	 *
+	 * @param event_type type of event we want to check (must be less than `PPM_EVENT_MAX`)
+	 * @return true if the event type has the `EC_METAEVENT` category.
+ */
+bool is_metaevent(ppm_event_code event_type);
+
+/**
+	 * @brief Return true if the event belongs to the `EC_UNKNOWN` category
+	 *
+	 * @param event_type type of event we want to check (must be less than `PPM_EVENT_MAX`)
+	 * @return true if the event type has the `EC_UNKNOWN` category.
+ */
+bool is_unknown_event(ppm_event_code event_type);
+
+/**
+	 * @brief Return true if the event belongs to the `EC_PLUGIN` category
+	 *
+	 * @param event_type type of event we want to check (must be less than `PPM_EVENT_MAX`)
+	 * @return true if the event type has the `EC_PLUGIN` category.
+ */
+bool is_plugin_event(ppm_event_code event_type);
+
+/*=============================== Events related ===============================*/
+
+/*=============================== PPM_SC set related (ppm_sc.cpp) ===============================*/
+
+/*!
+	\brief Provide the minimum set of syscalls required by `libsinsp` state collection.
+	If you call it without arguments it returns a new set with just these syscalls
+	otherwise, it merges the minimum set of syscalls with the one you provided.
+
+	WARNING: without using this method, we cannot guarantee that `libsinsp` state
+	will always be up to date, or even work at all.
+*/
+set<ppm_sc_code> enforce_sinsp_state_ppm_sc(set<ppm_sc_code> ppm_sc_of_interest = {});
```

Comment:
Since we are refactoring this bottom-up, I was thinking that maybe we could make these function names shorter, add the few ones missing, and be more consistent. Here's my proposal, but feel free to ignore it as this is more a matter of taste (the bold ones have not been implemented so far):
- all_sc_set
- all_tp_set
- **all_event_set**
- sinsp_state_sc_set
- sinsp_state_tp_set
- sinsp_state_event_set
- simple_sc_set
- io_sc_set
- io_sc_set
- file_sc_set
- net_sc_set
- proc_sc_set
- sys_sc_set
- sc_set_to_names
- tp_set_to_names
- event_set_to_names
- names_to_sc_set
- **names_to_tp_set**
- **names_to_event_set**
- sc_set_to_event_set
- **event_set_to_sc_set**

Reply:
Agree that `ppm` and `ppme` can be reserved as a concept under the hood, and in the API we use `sc` and `event` only. However, the `enforce` prefix would make it clearer that you can pass a set and add to it, so I would keep that.

Reply:
> However, the enforce prefix would make it clearer that you can pass a set and add to it, so I would keep that.

Given that new `libsinsp::events::set` class provides `merge` method, we could just "enforce" the set for `enforce_simple_ppm_sc_set`, leaving all other sets as just ... sets, without enforcing anything.
Libs client could do something like:
```
io_set = libsinsp::events::io_sc_set();
net_set = libsinsp::events::net_sc_set();

final_set = io_set.merge(net_set);
// use final set
```

Reply:
@jasondellaluce love the renamings! Will leave them as latest commit, once all build issues are fixes (macos... :/ )

Reply:
Love the approach of using `merge`! With that, most of these sets may also be made static so that we only need to built them at start time, and reuse them for all the runtime calls.

Reply:
Note: i'd avoid adding any new API for now; just rename existing ones; we can later add them as we need.
(eg: `libsinsp::events::all_event_set` would be new)

Reply:
Latest commit updated all function names!